packages/g/gamenetworkingsockets/xmake.lua

```diff
@@ -9,16 +9,36 @@ package("gamenetworkingsockets")
 
     add_versions("v1.2.0", "768a7cec2491e34c824204c4858351af2866618ceb13a024336dc1df8076bef3")
 
-    add_includedirs("include/")
+    on_load("windows", "linux", function(package)
+        if not package:config("shared") then
+            package:add("defines", "STEAMNETWORKINGSOCKETS_STATIC_LINK")
+            if is_plat("windows") then
+                package:add("deps", "libsodium", "protobuf-cpp")
+                package:add("syslinks", "ws2_32")
+            else
+                package:add("deps", "openssl", "protobuf-cpp", {configs = {cxflags = "-fpic"}})
```

Comment:
we can use `package:config("pic")` to add/fix pic in openssl/protobuf-cpp packages. 

```lua
package("protobuf-cpp")
    on_install(function (package)
        if not package:config("shared") and package:config("pic") then
            -- add -fpic flags
        end
    end)
```

Reply:
As far as I understand fPIC it has to be set on all .so libraries otherwise they cannot be loaded, maybe we should make this automatic?